olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp

```diff
@@ -0,0 +1,305 @@
+    network_ = std::make_shared<NiceMock<NetworkMock>>();
+
+    settings_.network_request_handler = network_;
+    settings_.cache = cache_;
+  }
+
+  void TearDown() override {
+    settings_.network_request_handler.reset();
+    settings_.cache.reset();
+
+    network_.reset();
+    cache_.reset();
+  }
+
+  std::shared_ptr<CacheMock> cache_;
+  std::shared_ptr<NetworkMock> network_;
+  olp::client::OlpClientSettings settings_;
+};
+
+TEST_F(CatalogRepositoryTest, GetLatestVersionCacheOnlyFound) {
+  olp::client::CancellationContext context;
+
+  auto request = DataRequest();
+
+  request.WithFetchOption(CacheOnly);
+
+  model::VersionResponse cached_vesrion;
+  cached_vesrion.SetVersion(10);
+
+  EXPECT_CALL(*cache_, Get(kCacheKey, _))
+      .Times(1)
+      .WillOnce(Return(cached_vesrion));
+
+  auto response = repository::CatalogRepository::GetLatestVersion(
+      kHRN, context, request, settings_);
+
+  ASSERT_TRUE(response.IsSuccessful());
+  EXPECT_EQ(10, response.GetResult().GetVersion());
+}
+
+TEST_F(CatalogRepositoryTest, GetLatestVersionCacheOnlyNotFound) {
+  olp::client::CancellationContext context;
+
+  auto request = olp::dataservice::read::DataRequest();
+
+  request.WithFetchOption(CacheOnly);
+
+  EXPECT_CALL(*cache_, Get(_, _)).Times(1).WillOnce(Return(boost::any{}));
+
+  ON_CALL(*network_, Send(_, _, _, _, _))
+      .WillByDefault([](olp::http::NetworkRequest, olp::http::Network::Payload,
+                        olp::http::Network::Callback,
+                        olp::http::Network::HeaderCallback,
+                        olp::http::Network::DataCallback) {
+        ADD_FAILURE() << "Should not be called with CacheOnly";
+        return olp::http::SendOutcome(olp::http::ErrorCode::UNKNOWN_ERROR);
+      });
+
+  auto response =
+      olp::dataservice::read::repository::CatalogRepository::GetLatestVersion(
+          kHRN, context, request, settings_);
+
+  EXPECT_FALSE(response.IsSuccessful());
+}
+
+TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyNotFound) {
+  olp::client::CancellationContext context;
+
+  auto request = olp::dataservice::read::DataRequest();
+
+  request.WithFetchOption(OnlineOnly);
+
+  ON_CALL(*cache_, Get(_, _))
+      .WillByDefault([](const std::string&, const olp::cache::Decoder&) {
+        ADD_FAILURE() << "Cache should not be used in OnlineOnly request";
+        return boost::any{};
+      });
+
+  EXPECT_CALL(*network_,
+              Send(IsGetRequest(OLP_SDK_URL_LOOKUP_METADATA), _, _, _, _))
+      .Times(1)
+      .WillOnce(NetworkMock::ReturnHttpResponse(
+          olp::http::NetworkResponse().WithStatus(
+              olp::http::HttpStatusCode::NOT_FOUND),
+          ""));
+
+  auto response =
+      olp::dataservice::read::repository::CatalogRepository::GetLatestVersion(
+          kHRN, context, request, settings_);
+
+  EXPECT_FALSE(response.IsSuccessful());
+}
+
+TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyFoundAndCacheWritten) {
```

Comment:
Looks like this test - `GetLatestVersionOnlineOnlyFoundAndCacheWritten` is more like integration test to me, cause we are testing interaction between `cache_` and `CatalogRepository`.

Thus, I would move it to the integration tests folder. Correct me if I'm wrong.

Reply:
Yes, it might be considered as integration test. But also it is simple unit test, which covers Write-Back routine of GetLatestVersion function. 
If you insist, I will move it.

Reply:
Ok, no need.